src/main/java/com/github/javafaker/Finance.java

```diff
@@ -99,75 +99,85 @@ private static String calculateIbanChecksum(String countryCode, String basicBank
 
     private static Map<String, String> createCountryCodeToBasicBankAccountNumberPatternMap() {
         // source: https://www.swift.com/standards/data-standards/iban
+        // version 87
         Map<String, String> ibanFormats = new HashMap<String, String>();
-        ibanFormats.put("AL", "\\d{8}[0-9A-Za-z]{16}");
         ibanFormats.put("AD", "\\d{4}\\d{4}[0-9A-Za-z]{12}");
+        ibanFormats.put("AE", "\\d{3}\\d{16}");
+        ibanFormats.put("AL", "\\d{8}[0-9A-Za-z]{16}");
         ibanFormats.put("AT", "\\d{5}\\d{11}");
         ibanFormats.put("AZ", "[A-Z]{4}[0-9A-Za-z]{20}");
-        ibanFormats.put("BH", "[A-Z]{4}[0-9A-Za-z]{14}");
-        ibanFormats.put("BE", "\\d{3}\\d{7}\\d{2}");
         ibanFormats.put("BA", "\\d{3}\\d{3}\\d{8}\\d{2}");
-        ibanFormats.put("BR", "\\d{8}\\d{5}\\d{10}[A-Z]{1}[0-9A-Za-z]{1}");
+        ibanFormats.put("BE", "\\d{3}\\d{7}\\d{2}");
         ibanFormats.put("BG", "[A-Z]{4}\\d{4}\\d{2}[0-9A-Za-z]{8}");
+        ibanFormats.put("BH", "[A-Z]{4}[0-9A-Za-z]{14}");
+        ibanFormats.put("BR", "\\d{8}\\d{5}\\d{10}[A-Z]{1}[0-9A-Za-z]{1}");
+        ibanFormats.put("BY", "[0-9A-Za-z]{4}\\d{4}[0-9A-Za-z]{16}");
+        ibanFormats.put("CH", "\\d{5}[0-9A-Za-z]{12}");
         ibanFormats.put("CR", "\\d{3}\\d{14}");
```

Comment:
For CR (Costa RIca) it has to contain a reserved digit `0`:
`ibanFormats.put("CR", "0\\d{3}\\d{14}");`
Check #674